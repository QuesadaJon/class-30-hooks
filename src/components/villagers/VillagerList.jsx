import React from 'react';
import PropTypes from 'prop-types';
import Villager from './Villager';
import { Link } from 'react-router-dom';

const VillagerList = ({ villagers }) => {
  const villagerElements = villagers.map((villager) => (
    <li key={villager._id}>
      <Link to={`/details/${villager._id}`}>
        <Villager 
          image={villager.image}
          name={villager.name}
          quote={villager.quote}
        />
      </Link>
    </li>
  ));

  return <ul data-testid="villagers">{villagerElements}</ul>;
};

VillagerList.propTypes = {
  villagers: PropTypes.arrayOf(
    PropTypes.shape({
      image: PropTypes.string.isRequired,
      name: PropTypes.string.isRequired,
      quote: PropTypes.string.isRequired,
    })
  ).isRequired,
};

export default VillagerList;
