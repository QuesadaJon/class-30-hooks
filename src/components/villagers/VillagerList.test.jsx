import React from 'react';
import { render } from '@testing-library/react';
import VillagerList from './VillagerList';

const villagers = [
  {
    _id: '1',
    image: 'https://image.com',
    name: 'name',
    quote: 'quote'
  },
  {
    _id: '3',
    image: 'https://image.com',
    name: 'name',
    quote: 'quote'
  },
  {
    _id: '2',
    image: 'https://image.com',
    name: 'name',
    quote: 'quote'
  }
];

describe('VillagerList component', () => {
  it('matches a snapshot', () => {
    const { asFragment } = render(<
      VillagerList
      villagers={villagers}
    />);

    expect(asFragment()).toMatchSnapshot();
  });
});
