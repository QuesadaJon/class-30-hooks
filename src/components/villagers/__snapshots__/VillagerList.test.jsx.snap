// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VillagerList component matches a snapshot 1`] = `
<DocumentFragment>
  <ul
    data-testid="villagers"
  >
    <li>
      <a
        href="/details/1"
      >
        <figure
          class="Villager"
        >
          <p>
            name
          </p>
          <img
            alt="name"
            src="https://image.com"
          />
          <figcaption>
            <p>
              quote
            </p>
          </figcaption>
        </figure>
      </a>
    </li>
    <li>
      <a
        href="/details/3"
      >
        <figure
          class="Villager"
        >
          <p>
            name
          </p>
          <img
            alt="name"
            src="https://image.com"
          />
          <figcaption>
            <p>
              quote
            </p>
          </figcaption>
        </figure>
      </a>
    </li>
    <li>
      <a
        href="/details/2"
      >
        <figure
          class="Villager"
        >
          <p>
            name
          </p>
          <img
            alt="name"
            src="https://image.com"
          />
          <figcaption>
            <p>
              quote
            </p>
          </figcaption>
        </figure>
      </a>
    </li>
  </ul>
</DocumentFragment>
`;
